slamon_afm/tests/models_tests.py

```diff
@@ -160,6 +162,209 @@ def testDontClaimErrorTasks(self):
         )
         db.session.add(task)
         db.session.flush()
-
         claimed = list(Task.claim_tasks(self.agent, 1))
         self.assertEqual(len(claimed), 0)
+
+
+class TaskStatsTests(AFMTest):
+    def setUp(self):
+        super(TaskStatsTests, self).setUp()
+        self.agent = Agent(
+            uuid='de305d54-75b4-431b-adb2-eb6b9e546013',
+            name='test agent',
+            last_seen=datetime.utcnow(),
+            capabilities=[AgentCapability(type='task-type-1', version=1)]
+        )
+        db.session.add(self.agent)
+        db.session.flush()
+        self.assertEqual(db.session.query(Agent).count(), 1)
+
+    def _addTask(self, **kwargs):
+        default = {
+            'uuid': 'de305d54-75b4-431b-adb2-eb6b9e546013',
+            'test_id': 'de305d54-75b4-431b-adb2-eb6b9e546013',
+            'type': 'task-type-1',
+            'version': 1,
+            'data': {}
+        }
+        default.update(kwargs)
+        task = Task(**default)
+        db.session.add(task)
+        db.session.flush()
+        return task
+
+    @patch.object(statsd, 'incr', return_value=None)
+    @patch.object(statsd, 'timing', return_value=None)
+    @patch.object(statsd, 'gauge', return_value=None)
+    def testIncPostedStats(self, gauge_mock, timing_mock, incr_mock):
+        self._addTask()
+
+        incr_mock.assert_called_once_with('tasks.posted')
+        timing_mock.assert_not_called()
+        gauge_mock.assert_has_calls([
+            call('tasks.task-type-1.1.queue', 1),
+            call('tasks.task-type-1.1.processing', 0)
+        ], any_order=True)
+
+    def testIncDeletedStats(self):
+        task = self._addTask()
+        with patch.object(statsd, 'incr', return_value=None) as incr_mock, \
+                patch.object(statsd, 'timing') as timing_mock:
+            db.session.delete(task)
+            db.session.flush()
+
+            incr_mock.assert_called_once_with('tasks.deleted')
+            timing_mock.assert_called_once_with('tasks.task-type-1.1.deleted', ANY)
+
+    def testIncClaimedStats(self):
+        self._addTask()
+
+        # create list to iterate the generator returned by claim_tasks
+        with patch.object(statsd, 'incr', return_value=None) as incr_mock, \
+                patch.object(statsd, 'timing') as timing_mock,\
+                patch.object(statsd, 'gauge') as gauge_mock:
+            _ = list(Task.claim_tasks(self.agent, 1))
```

Comment:
What is this `_` variable? I don't see a reference to it anywhere after this... You certain this assignment is necessary?